projects/dsp-ui/src/lib/core/session.service.ts

```diff
@@ -0,0 +1,215 @@
+import { Inject, Injectable } from '@angular/core';
+import {
+    ApiResponseData,
+    ApiResponseError,
+    Constants,
+    CredentialsResponse,
+    KnoraApiConfig,
+    KnoraApiConnection,
+    UserResponse
+} from '@dasch-swiss/dsp-js';
+import { DspApiConfigToken, DspApiConnectionToken } from './core.module';
+
+/**
+ * Currently logged-in user information
+ */
+export interface CurrentUser {
+    // username
+    name: string;
+
+    // json web token
+    jwt: string;
+
+    // default language for ui
+    lang: string;
+
+    // is system admin?
+    sysAdmin: boolean;
+
+    // list of project shortcodes where the user is project admin
+    projectAdmin: string[];
+}
+
+/**
+ * Session with id (= login timestamp) and inforamtion about logged-in user
+ */
+export interface Session {
+    id: number;
+    user: CurrentUser;
+}
+
+@Injectable({
+    providedIn: 'root'
+})
+export class SessionService {
+
+    /**
+     * max session time in milliseconds
+     * default value (24h): 86400000
+     *
+     */
+    readonly MAX_SESSION_TIME: number = 86400000; // 1d = 24 * 60 * 60 * 1000
+
+
+    constructor(
+        @Inject(DspApiConnectionToken) private dspApiConnection: KnoraApiConnection,
+        @Inject(DspApiConfigToken) private dspApiConfig: KnoraApiConfig
+    ) { }
+
+    /**
+     * get session information from localstorage
+     */
+    getSession(): Session {
+        if (localStorage.getItem('session') !== null) {
+            return JSON.parse(localStorage.getItem('session'));
+        }
+
+        return undefined;
```

Comment:
functions return `undefined` by default, no need for this.

Reply:
better return `null`

Reply:
Resolved in cb139e0